### PR TITLE
rose edit: use multiline widget for newline value

### LINKED ISSUE #1659: rose edit: use multiline widget if newlines in entry box
A `gtk.Entry` can display weird characters in place of new lines - if a variable value has newlines in it (and no other widget-related information is given), `rose edit` should use a multiline text widget for that variable value.

### PR DESCRIPTION
**Pull request by benfitzpatrick**
This fixes #1659.

@matthewrmshin, please review 1.
@arjclark, please review 2.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Looks OK 1.

I guess there is always a chance that only a single line value is specified for a value that can accept multiple lines?

File: demo/rose-config-edit/demo_meta/app/01-types/rose-app.conf [hunk 1]
```diff
@@ -33,6 +33,8 @@ my_python_list=["Foo",5,False,"banana"]
 my_quoted="Double quoted string, allows \" with backslash"
 my_quoted_array="a","b","c"
 my_raw='Raw string'
+my_raw_with_newlines=Doe
+                    =a deer
```
Comment (arjclark) [inline demo/rose-config-edit/demo_meta/app/01-types/rose-app.conf]: =a female deer ?

Comment (benfitzpatrick) [inline demo/rose-config-edit/demo_meta/app/01-types/rose-app.conf]: That's what I added in `rose edit` to test the widget!

**Comment by benfitzpatrick** [comment]
They can add proper multiline support via the `widget[...]` metadata option.

There is a reference to type = 'str_multi' in the code, but that is not actually valid according to metadata-check, or documented anywhere... worth adding as a valid type?

**Comment by arjclark** [comment]
In the test example it seems to render a redundant extra (uneditable) line in the gui after the last line entry e.g.

```
doe
a deer

```

seems a bit confusing as I'd expected to be able to do something with that?

**Comment by arjclark** [comment]
@benfitzpatrick - awaiting your response on the above comment.

**Comment by benfitzpatrick** [comment]
Looks like the way the scrolled window widget works with a textview - I'll have a look.

**Comment by benfitzpatrick** [comment]
Too ancient to remain open as a pull request - will revisit.

**Comment by matthewrmshin** [comment]
(Not really a *wontfix*, but more like a *will fix* when we have time.)

**Pull request closed without merging**